docs/DeveloperGuide.md

```diff
@@ -1,38 +1,239 @@
-# Developer Guide
-
-## Acknowledgements
-
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
-
-## Design & implementation
-
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
-
-
-## Product scope
-### Target user profile
-
-{Describe the target user profile}
-
-### Value proposition
-
-{Describe the value proposition: what problem does it solve?}
-
-## User Stories
-
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
-
-## Non-Functional Requirements
-
-{Give non-functional requirements}
-
-## Glossary
-
-* *glossary item* - Definition
-
-## Instructions for manual testing
-
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+---
+title: Developer Guide
+---
+
+* Table of Contents
+{:toc}
+
+# Developer Guide
+
+## Acknowledgements
+
+* The CS2113 Team (Professor and Teaching Assistants) For their guidance in teaching us concepts of Software
+  Engineering and OOP
+
+* Exercise Data retrieved from: [Wrkout](https://github.com/wrkout/exercises.json) (open source MIT license)
+  Used to generate the data.json file containing all the exercises data used
+
+## Design & implementation
+
+### Architecture Component
+
+<div align="center">
+<img src="UML/Images/architecture.png"/>
+<p>
+Figure 1.1
+</p>
+</div>

```

Comment:
Not too sure why there are dotted arrows and solid arrows in the diagram, could be clearer in explaining the difference as developers might not understand the meaning behind them

![image](https://user-images.githubusercontent.com/74498975/228740381-b0b6fc00-f67e-4d0a-9deb-969d006956bc.png)